### ISSUE TITLE
tnt.boom: lack of central on_blast call

### ISSUE THREAD

**Issue by DS-Minetest** [minetest/minetest_game#1825]
It should call the node's `on_blast` and drop its drop. Here: https://github.com/minetest/minetest_game/blob/master/mods/tnt/init.lua#L388 Maybe check a `delete_center` field of `def` because of exploding nodes that shouldn't drop their drops.

**Comment by cx384** [comment]
How did you noticed that?

**Comment by paramat** [comment]
The thread title is unclear, deleting is not the problem, it's the lack of central on_blast call and drops.
I guess it is like this because it is usually TNT at the centre, so is this issue due to custom use of tnt_boom()?

> Maybe check a delete_center field to def because of exploding nodes that shouldn't drop their drops.

I don't see a need for that yet.

**Comment by DS-Minetest** [comment]
>How did you noticed that?

I shot with the handcannon from technic_fight, which uses `tnt.boom` on impact, on technic's forcefield when it wasn't resistant against blasts yet. I wanted to see how fast I have to shoot to come through. I hit the forcefield emitter and recognized that the forcefield nodes weren't removed. So, I added an `on_blast` to the forcefield emitter which removes the forcefield nodes. But as I shot at the forcefield emitter again it didn't work and I wondered what I did wrong. After some time I recognized that nothing was dropped but the function returns the disabled emitter. So, I shot near the forcefield emitter and it worked. When I looked into tnt code I saw that at the center pos the blast node was set and everything was clear.

>The thread title is unclear, deleting is not the problem, it's the lack of central on_blast call and drops.

I'll change it.
>I guess it is like this because it is usually TNT at the centre, so is this issue due to custom use of tnt_boom()?

Yes.
>>Maybe check a delete_center field to def because of exploding nodes that shouldn't drop their drops.

>I don't see a need for that yet.

Tnt uses `tnt.boom`. And also has a `tnt.boom` in `on_blast`. If every `tnt.boom` would call the central `on_blast`, tnt would cause an infinite loop.
